ssz/simple-serialize.md

```diff
@@ -152,6 +160,31 @@ is equal to the default value for that type.
   count of `1` than fields are illegal.
 - The `None` type option in a `Union` type is only legal as the first option
   (i.e. with index zero).
+- `CompatibleUnion({})` without any type options are illegal.
+- `CompatibleUnion({selector: type})` with a selector outside `uint8(1)` through
+  `uint8(127)` are illegal.
```

Comment:
Shouldn't this allow `0` as a valid selector? `[0..127]`.

Reply:
https://eips.ethereum.org/EIPS/eip-8016#why-are-compatibleunion-selectors-limited-to-1--127

Currently reserving 0 for two reasons:
- Preventing bugs from default zero initialization (require selection of a specific case)
- Allow future expansion into None values (right now not needed)

The selector does not indicate an offset, so does not have a base, one might use values 5 and 12 without 1 through 4.